### PR TITLE
feat: Implement data source and layer context

### PR DESCRIPTION
**Pull request by msasinowski**

### CODE REVIEW

File: src/components/AzureMapFeature/AzureMapFeature.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { AzureMapDataSourceContext } from '../../contexts/AzureMapDataSourceContext'
+
+const createFeature = ({
+  type,
+  coordinate,
+  coordinates,
+  multipleCoordinates,
+  multipleDimensionCoordinates,
+  bbox
+}: IAzureMapFeature): atlas.data.Geometry | undefined => {
```
Comment (srednicki95) [inline src/components/AzureMapFeature/AzureMapFeature.tsx]: undefined?

File: src/contexts/AzureMapLayerContext.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      }
+    }
+  }, [])
+
+  return (
+    <Provider
+      value={{
+        layerRef
+      }}
+    ></Provider>
```
Comment (srednicki95) [inline src/contexts/AzureMapLayerContext.tsx]: Is this Layer provider will be without any children?

File: src/components/AzureMapFeature/AzureMapFeature.tsx (hunk 1 above)
Comment (msasinowski) [reply src/components/AzureMapFeature/AzureMapFeature.tsx]: It is coming from helper function, we can make it as null

File: src/contexts/AzureMapLayerContext.tsx (hunk 2 above)
Comment (msasinowski) [reply src/contexts/AzureMapLayerContext.tsx]: As I see in the previous implementation layer was used only to add the DataSource, we can adjust the implementation if we needed to

**Pull request merged**